aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -206,25 +231,44 @@ module aptos_framework::fungible_asset {
         TransferRef { metadata }
     }
 
+    /// Creates a extend ref that can be used to upgrade to concurrent from the given fungible
+    /// object's constructor ref.
+    /// This can only be called at object creation time as constructor_ref is only available then.
+    public fun generate_extend_ref(constructor_ref: &ConstructorRef): ExtendRef {
+        object::generate_extend_ref(constructor_ref)
+        // let metadata = object::object_from_constructor_ref<Metadata>(constructor_ref);
+        // ExtendRef { metadata }
```

Comment:
nits

Reply:
I was wondering, why do we do, is there a reason:

let metadata = object::object_from_constructor_ref<Metadata>(constructor_ref);
        TransferRef { metadata }
       
above, instead of object::generate_transfer_ref(constructor_ref)?

Reply:
wait, this has to be fixed...
`object::generate_extend_ref(constructor_ref)` return `object::ExtendRef`.
`TransferRef` here is `fungible_asset::TransferRef`.
cc @davidiw ... proof that caps is a better name than refs here cuz ppl will be confused......

Reply:
so we have two different things referred to as TransferRef, that mean different things ?  it might have been better to name fungible_asset::TransferRef as AssetTransferRef or something?

but that means that I don't need this function at all, I will just directly call object::generate_extend_ref in the test (and I assume creators of fungible assets can do the same)

Reply:
yeah, this one seems unnecessary to me cuz you could just get it from `object::generate_extend_ref`.
`TransferRef` here means transfer FA between FungibleStore whileas `object::TransferRef` is for transferring objects.
I named it as `TransferCap` initially, but DW thought it is better to be consistent with object module.

Reply:
Updated